Add fingerprint and car info for 2021 Kia Niro PHEV

dblacknc opened the pull request:
<!--- ***** Template: Car Fingerprint *****

**Checklist**

**Car**
2021 Kia Niro PHEV

**Route**
3d264cee10fdc8d3/2024-02-02--05-50-49

-->

This car is almost identical to the 2022 Kia Niro PHEV, except it has a couple different fingerprint values, and uses a different Hyundai harness.  Broke the '21 and 22's values into a separate set, with correct mass and steeringRatio (vs. averages over many Niros as the '22 was previously included).

dblacknc committed:
Add 2021 Kia Niro Plug-in Hybrid

dblacknc committed:
Added CAR.KIA_NIRO_PHEV_2021

dblacknc committed:
Added CAR.KIA_NIRO_PHEV_2021

Created new block and moved 2022 model to it as well, with correct mass and steerRatio values.

dblacknc committed:
Added 2021 Kia Niro PHEV

dblacknc committed:
Added 2021 Kia Niro PHEV

For now, using 2020 Niro EV torque data

dblacknc committed:
Update values.py

Add '21 Kia Niro PHEV to HYBRID_CAR for gas signal and DBC info

dblacknc committed:
Test route for 2021 Kia Niro PHEV

sshane commented on selfdrive/car/hyundai/values.py:
```diff
@@ -113,6 +113,7 @@ class CAR(StrEnum):
   KIA_NIRO_EV = "KIA NIRO EV 2020"
   KIA_NIRO_EV_2ND_GEN = "KIA NIRO EV 2ND GEN"
   KIA_NIRO_PHEV = "KIA NIRO HYBRID 2019"
+  KIA_NIRO_PHEV_2021 = "KIA NIRO PLUG-IN HYBRID 2021"
```
Any reason this can't be KIA_NIRO_PHEV_2022? We'd prefer to combine platforms as it improves fingerprinting, and then fall back to detecting features.

dblacknc replied:
The 2021 and 22 use different harnesses and I didn't see a clean way to make them the same car.   As my changes propose, I would at least like the 2021 and 22 broke out as a car separate from earlier years of the PHEV, because there's at least one feature like minimum cruise control speed (0 mph for the above vs. 32 for 2020 and earlier), and the car mass is much lower than the average that includes earlier EV models.

dblacknc replied:
BTW now I don't know why the new provided test route is failing.   It's uploaded, as I see it in Connect, and I marked it public and preserved.

dblacknc replied:
BTW I don't know why the test route check is failing.  It's preserved and public.  In useradmin the route name has a forward slash in it and I tried changing it (the typo fix) to vertical bar - still failed.

sshane replied:
You can just made the car info value a list.

dblacknc replied:
OK, I think I understand and will go back and do that.  Any idea about the test route fail, or is that not important?

dblacknc committed:
Add public test route for 2021 Kia Niro PHEV

dblacknc committed:
Fixed typo in test route for 2021 Kia Niro PHEV

dblacknc committed:
Move '21 Kia Niro PHEV car info under '22

...and remove other mention of '21 as distinct car

dblacknc committed:
Added '21 Kia Niro PHEV fingerprint under '22

...and removed separate car definition

dblacknc committed:
Removed '21 Kia Niro PHEV as separate car, now under '22

dblacknc committed:
Removed '21 Kia Niro PHEV, now under '22 fingerprint

dblacknc committed:
Removed test route for '21 Kia Niro PHEV

dblacknc committed:
Merge branch 'commaai:master' into master

dblacknc committed:
Update values.py

Missed removing one more reference to '21 Kiro Niro PHEV, consolidating under '22.

dblacknc committed:
Merge branch 'commaai:master' into master

sshane commented on selfdrive/car/hyundai/interface.py:
```diff
@@ -350,7 +355,7 @@ def _update(self, c):
       ret.buttonEvents = create_button_events(self.CS.cruise_buttons[-1], self.CS.prev_cruise_buttons, BUTTONS_DICT)
 
-    # On some newer model years, the CANCEL button acts as a pause/resume button based on the PCM state
+    # On some newer model years, the CANCEL button acts as a pause/resume button based on the PCM stateUpdate fingerprints.py
```
```suggestion
    # On some newer model years, the CANCEL button acts as a pause/resume button based on the PCM state
```

sshane committed:
Update selfdrive/car/hyundai/interface.py

sshane commented on selfdrive/car/hyundai/interface.py:
```diff
@@ -183,13 +183,18 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
       ret.tireStiffnessFactor = 0.385
       if candidate == CAR.KIA_NIRO_PHEV:
         ret.minSteerSpeed = 32 * CV.MPH_TO_MS
+    elif candidate == CAR.KIA_NIRO_PHEV_2022:
+      ret.mass = 3384. * CV.LB_TO_KG
+      ret.wheelbase = 2.7
+      ret.steerRatio = 13.3
+      ret.tireStiffnessFactor = 0.385
```
```suggestion
```

sshane commented on selfdrive/car/hyundai/interface.py:
```diff
@@ -183,13 +183,18 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
       ret.wheelbase = 2.78
       ret.steerRatio = 14.4 * 1.1   # 10% higher at the center seems reasonable
-    elif candidate in (CAR.KIA_NIRO_EV, CAR.KIA_NIRO_EV_2ND_GEN, CAR.KIA_NIRO_PHEV, CAR.KIA_NIRO_HEV_2021, CAR.KIA_NIRO_HEV_2ND_GEN, CAR.KIA_NIRO_PHEV_2022):
+    elif candidate in (CAR.KIA_NIRO_EV, CAR.KIA_NIRO_EV_2ND_GEN, CAR.KIA_NIRO_PHEV, CAR.KIA_NIRO_HEV_2021, CAR.KIA_NIRO_HEV_2ND_GEN):
```
```suggestion
    elif candidate in (CAR.KIA_NIRO_EV, CAR.KIA_NIRO_EV_2ND_GEN, CAR.KIA_NIRO_PHEV, CAR.KIA_NIRO_HEV_2021, CAR.KIA_NIRO_HEV_2ND_GEN, CAR.KIA_NIRO_PHEV_2022):
```

sshane commented on selfdrive/car/hyundai/fingerprints.py:
```diff
@@ -1199,6 +1199,7 @@
+      b'\xf1\x816H6G5051\x00\x00\x00\x00\x00\x00\x00\x00',
     ],
     (Ecu.transmission, 0x7e1, None): [
       b'\xf1\x006U3H1_C2\x00\x006U3J9051\x00\x00PDE0G16NL3\x00\x00\x00\x00',
```
```suggestion
      b'\xf1\x006U3H1_C2\x00\x006U3J9051\x00\x00PDE0G16NL3\x00\x00\x00\x00',
      b'\xf1\x816U3J9051\x00\x00\xf1\x006U3H1_C2\x00\x006U3J9051\x00\x00PDE0G16NL3\x00\x00\x00\x00',
```

sshane committed:
Update selfdrive/car/hyundai/fingerprints.py

sshane committed:
Apply suggestions from code review

sshane commented:
The average curb weight for the 2021 hybrids are 3163.5 lbs, and the 2021 plug-in hybrids are 3413.5 lbs, @nuwandavek would you say these differences aren't significant enough to require two platforms?
